File: 4h-unfolder-win/src/FourHUnfolder.Geometry/Algorithms/EdgeMarker.cs
using FourHUnfolder.Domain.Models;

namespace FourHUnfolder.Geometry.Algorithms;

/// <summary>
/// Stamps EdgeType on every mesh edge based on whether it's in the MST.
///   MST interior edge  → Fold
///   Non-MST interior   → Cut
///   Boundary edge      → Boundary
/// </summary>
public class EdgeMarker
{
    public void Mark(Mesh mesh, IReadOnlySet<int> foldEdgeIds)
    {
        foreach (var edge in mesh.Edges)
        {
            edge.Type = edge.ConnectsFaces
                ? (foldEdgeIds.Contains(edge.Id) ? EdgeType.Fold : EdgeType.Cut)
                : EdgeType.Boundary;
        }
    }
}

File: 4h-unfolder-mac-swift/Sources/FourHUnfolderCore/Core/Algorithms/EdgeMarker.swift
// MARK: - Stamps EdgeType on every mesh edge after MST + overrides are resolved

struct EdgeMarker {
    func mark(mesh: Mesh, foldEdgeIds: Set<Int>) {
        for i in mesh.edges.indices {
            let e = mesh.edges[i]
            if e.connectsFaces {
                mesh.edges[i].type = foldEdgeIds.contains(e.id) ? .fold : .cut
            } else {
                mesh.edges[i].type = .boundary
            }
        }
    }
}
